linkml/generators/pythongen.py

```diff
@@ -126,14 +127,135 @@ def visit_type(self, typ: TypeDefinition) -> None:
             if type_prefix:
                 self.emit_prefixes.add(type_prefix)
 
+    def _import_field(self):
+        enum_pvs = []
+        for enum_name, enum_def in self.schema.enums.items():
+            for pv in enum_def.permissible_values:
+                enum_pvs.append(f"{enum_name}({pv})")
+        for _, class_def in self.schema.classes.items():
+            for slot in class_def.slots:
+                if self.schema.slots[slot].ifabsent in enum_pvs:
+                    return True
+        return False
+
     def gen_schema(self) -> str:
+        all_imports = Imports()
+        # generic imports
+        all_imports = (
+            all_imports
+            + Import(module="dataclasses")
+            + Import(module="re")
+            + Import(
+                module="jsonasobj2",
+                objects=[
+                    ObjectImport(name="JsonObj"),
+                    ObjectImport(name="as_dict"),
+                ],
+            )
+            + Import(
+                module="typing",
+                objects=[
+                    ObjectImport(name="Optional"),
+                    ObjectImport(name="List"),
+                    ObjectImport(name="Union"),
+                    ObjectImport(name="Dict"),
+                    ObjectImport(name="ClassVar"),
+                    ObjectImport(name="Any"),
+                ],
+            )
+            + Import(
+                module="dataclasses",
+                objects=[
+                    ObjectImport(name="dataclass"),
+                ],
+            )
+            + Import(
+                module="datetime",
+                objects=[
+                    ObjectImport(name="date"),
+                    ObjectImport(name="datetime"),
+                    ObjectImport(name="time"),
+                ],
+            )
+        )
+
         # The metamodel uses Enumerations to define itself, so don't import if we are generating the metamodel
-        enumimports = (
-            ""
-            if self.genmeta
-            else "from linkml_runtime.linkml_model.meta import EnumDefinition, PermissibleValue, PvFormulaOptions\n"
+        if not self.genmeta:
```

Comment:
eg. could do this

at a module level...

```python
DEFAULT_IMPORTS = Imports(imports=[
    Import(module="dataclasses", objects=[ObjectImport(name="dataclass")]),
    # ... the rest of the default imports
])

METAMODEL_IMPORTS = Imports(imports=[
    Import(module="linkml_runtime.linkml_model.meta", objects=[ObjectImport(name="EnumDefinition")]
    # ... etc.
])

# and any other import groups
```

and then in the generator you could do

```python
def gen_schema(self):
    imports = DEFAULT_IMPORTS.model_copy()
    if not self.genmeta:
        imports += METAMODEL_IMPORTS
```

Reply:
Fully agree here too. But refactoring that I would rather address in a separate PR to bring this one to an end.

Reply:
i think it would take approximately 10 seconds, but i don't use the python generator so it doesn't really matter that much to me

Reply:
My experience is that you know when you start refactoring, but you never know when you will end 😉

And I have the feeling that my PRs tend to get much too big and extensive over time to be digestible (see also #2303, with 104 comments!). That is why I am trying to close this one before it gets into another [Berlin Brandenburg Airport](https://en.wikipedia.org/wiki/Construction_of_Berlin_Brandenburg_Airport).

Reply:
sure sure, do whatever, like i said i don't use the python generator so idrc <3

Reply:
This should be also addresses on #2367.